samples/python/winml_tutorial/pywinrt_output/pyrt/src/py.Windows.AI.MachineLearning.cpp

```diff
@@ -0,0 +1,6732 @@
+        if (!out_return_value) 
+        { 
+            return nullptr;
+        }
+        
+        return out_return_value.detach();
+    }
+    catch (...)
+    {
+        return py::to_PyErr();
+    }
+}
+
+static PyObject* __LearningModelEvaluationResult_from(PyObject* /*unused*/, PyObject* arg)
+{
+    try
+    {
+        auto instance = py::converter<winrt::Windows::Foundation::IInspectable>::convert_to(arg);
+        return py::convert(instance.as<winrt::Windows::AI::MachineLearning::LearningModelEvaluationResult>());
+    }
+    catch (...)
+    {
+        return py::to_PyErr();
+    }
+}
+
+static PyMethodDef LearningModelEvaluationResult_methods[] = {
+    { "_from", (PyCFunction)__LearningModelEvaluationResult_from, METH_O | METH_STATIC, nullptr },
+    { nullptr }
+};
+
+static PyGetSetDef LearningModelEvaluationResult_getset[] = {
+    { const_cast<char*>("CorrelationId"), (getter)LearningModelEvaluationResult_get_CorrelationId, nullptr, nullptr, nullptr },
+    { const_cast<char*>("ErrorStatus"), (getter)LearningModelEvaluationResult_get_ErrorStatus, nullptr, nullptr, nullptr },
+    { const_cast<char*>("Outputs"), (getter)LearningModelEvaluationResult_get_Outputs, nullptr, nullptr, nullptr },
+    { const_cast<char*>("Succeeded"), (getter)LearningModelEvaluationResult_get_Succeeded, nullptr, nullptr, nullptr },
+    { nullptr }
+};
+
+static PyType_Slot LearningModelEvaluationResult_Type_slots[] = 
+{
+    { Py_tp_dealloc, LearningModelEvaluationResult_dealloc },
+    { Py_tp_new, LearningModelEvaluationResult_new },
+    { Py_tp_methods, LearningModelEvaluationResult_methods },
+    { Py_tp_getset, LearningModelEvaluationResult_getset },
+    { 0, nullptr },
+};
+
+static PyType_Spec LearningModelEvaluationResult_Type_spec =
+{
+    "_pyrt_Windows_AI_MachineLearning.LearningModelEvaluationResult",
+    sizeof(py::winrt_wrapper<winrt::Windows::AI::MachineLearning::LearningModelEvaluationResult>),
+    0,
+    Py_TPFLAGS_DEFAULT,
+    LearningModelEvaluationResult_Type_slots
+};
+
+// ----- LearningModelSession class --------------------
+
+PyTypeObject* py::winrt_type<winrt::Windows::AI::MachineLearning::LearningModelSession>::python_type;
+
+static PyObject* LearningModelSession_new(PyTypeObject* type, PyObject* args, PyObject* kwds)
+{
+    if (kwds != nullptr)
+    {
+        PyErr_SetString(PyExc_TypeError, "keyword arguments not supported");
+        return nullptr;
+    }
+    
+    Py_ssize_t arg_count = PyTuple_Size(args);
+    
+    if (arg_count == 1)
+    {
+        try
+        {
+            auto param0 = py::convert_to<winrt::Windows::AI::MachineLearning::LearningModel>(args, 0);
+            
+            winrt::Windows::AI::MachineLearning::LearningModelSession instance{ param0 };
+            return py::wrap(instance, type);
+        }
+        catch (...)
+        {
+            return py::to_PyErr();
+        }
+    }
+    else if (arg_count == 2)
+    {
+        try
```

Comment:
this is a good sample to comment on about a few things, as we're trying to reduce image size:

- consider a py::try_create(create_func) to consolidate all the try/catch blocks, wrapping the incoming create_func lambda 
- consider passing PyObject * args as a PyConvertibleObject * args, providing implicit conversion so that args[n] can be passed directly as a param instead of requiring explicit convert_to calls
-

Reply:
Please comment on #217